enhancements/machine-config/on-cluster-builds.md

```diff
@@ -0,0 +1,1009 @@
+| ------------------------------------------------- | -------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `.spec.customContent.dockerfile`                  | Optional                         | The Dockerfile intended for use by the BuildController in order to build a customized OS image.                                                                                                                                                                                                            |
+| `.spec.customContent.imageRegistryPushSecretName` | Required if `dockerfile` is set. | Reference to a Kubernetes secret that contains the required registry credentials with sufficient permission to push the image.                                                                                                                                                                             |
+| `.spec.customContent.imageRegistryPullSecretName` | Required if `dockerfile` is set. | Reference to a Kubernetes secret that contains the required registry credentials with sufficient permission to pull both the base image as well as the final OS image. When absent, the global pull secret will be used.                                                                                   |
+| `.spec.customContent.imagePushTarget`             | Required if `dockerfile` is set. | Where to push the resulting image build, e.g. `registry.host.com/org/repo:tag`. The tag will be ignored.                                                                                                                                                                                                   |
+| `.spec.customContent.deployMode`                  | Optional                         | Controls how the MachineConfigs should be combined with the custom OS contents. Possible options are: `image` and `bootc`. See Implementation Details for details.                                                                                                                                         |
+| `.spec.customContent.baseImagePullspec`           | Optional                         | The fully-qualified image pullspec of the base OS image to use. When empty, it will default to the default OS image pullspec for the current OpenShift release. When populated, it must be pullable using either the global pull secret or the credential referenced by imageRegistryPullSecretName above. |
```

Comment:
Is this not included in the dockerfile? If  it is included, then we should specify why it is needed here as well.